packages/core/src/components/non-ideal-state/nonIdealState.tsx

```diff
@@ -79,34 +79,20 @@ export interface NonIdealStateProps extends Props {
  *
  * @see https://blueprintjs.com/docs/#core/components/non-ideal-state
  */
-export class NonIdealState extends AbstractPureComponent<NonIdealStateProps> {
-    public static displayName = `${DISPLAYNAME_PREFIX}.NonIdealState`;
-
-    public static defaultProps: Partial<NonIdealStateProps> = {
-        iconMuted: true,
-        iconSize: NonIdealStateIconSize.STANDARD,
-        layout: "vertical",
-    };
-
-    public render() {
-        const { action, children, className, layout } = this.props;
-
+export const NonIdealState: React.FC<NonIdealStateProps> = ({
+    action,
+    children,
+    className,
+    description,
+    icon,
+    iconMuted = true,
+    iconSize = NonIdealStateIconSize.STANDARD,
+    layout = "vertical",
+    title,
+}) => {
```

Comment:
Can we destructure these inside of the function body rather than in the arguments? I think this is more in line with the structure we use for other functional components that have multiline props in Blueprint.

```tsx
export const NonIdealState: React.FC<NonIdealStateProps> = props => {
    const {
        action,
        children,
        className,
        description,
        icon,
        iconMuted = true,
        iconSize = NonIdealStateIconSize.STANDARD,
        layout = "vertical",
        title,
    } = props;
    // ...
}
```